web/src/models/RequiredGatesResult.model.ts

```diff
@@ -0,0 +1,22 @@
+import {Model, TTestRunnerSchemas} from 'types/Common.types';
+
+export type TRawRequiredGatesResult = TTestRunnerSchemas['RequiredGatesResult'];
+type RequiredGatesResult = Model<
+  TRawRequiredGatesResult,
+  {
+    requiredFailedGates: string[];
+  }
+>;
+
+const RequiredGatesResult = ({
+  required = [],
+  failed = [],
+  passed = true,
+}: TRawRequiredGatesResult): RequiredGatesResult => ({
+  required,
+  failed,
+  passed,
+  requiredFailedGates: required.filter(gate => failed.includes(gate)),
```

Comment:
Calculates the failed gates that are required from the model so we can use them elsewhere